App/PyImports/Utils/Helpers.py

```diff
@@ -0,0 +1,38 @@
+def dict2xml(d, root_node=None):
+    wrap          =  False if None == root_node or isinstance(d, list) else True
+    root          = 'root' if None == root_node else root_node
+    root_singular = root[:-1] if 's' == root[-1] and None == root_node else root
+    xml           = ''
+    attr          = ''
+    children      = []
+
+    if isinstance(d, dict):
+        for key, value in dict.items(d):
+            if isinstance(value, dict):
+                children.append(dict2xml(value, key))
+            elif isinstance(value, list):
+                children.append(dict2xml(value, key))
+            elif key[0] == '@':
+                attr = attr + ' ' + key[1::] + '="' + str(value) + '"'
+            else:
+                xml = '<' + key + ">" + str(value) + '</' + key + '>'
+                children.append(xml)
+
+    else:
+        #if list
+        for value in d:
+            children.append(dict2xml(value, root_singular))
+
+    end_tag = '>' if 0 < len(children) else '/>'
+
+    if wrap or isinstance(d, dict):
```

Comment:
`wrap` already contains conditional for isinstance(d, dict)

Reply:
`wrap` contains` isinstance(d, list)`, not `isinstance(d, dict)`